### DISCUSSION TITLE
Can I rename a Parameter?

### DISCUSSION THREAD

**Discussion by rowlesmr** [lmfit/lmfit-py#867] [Q&A] [answered]
Is it possible to rename a Parameter?

.

I have a non-linear system with five parameters, where it makes sense to have the parameter values ordered such that `a<b<c<d<e`. I don't want to constrain the values, as it is possible to get to a valid solution, and have the values unordered. You just then do some post-processing, and reorder them after the fact.

This after-the-fact reordering of the values is what I was doing previously, but now that I've ironed out a lot of the kinks, I'd like to be a bit more robust in my analysis of potential solutions, so I'd like a more robust way of keeping all the metadata together.

**Comment by newville** [comment]
@rowlesmr  I think the answer is No.    Parameter names correspond to the mathematical symbol in the function generating the residual of the fit.   Renaming Parameters during the fit is not possible.  Renaming them after the fact would deliberately alter the meaning of the fit.   I must not understand what you are trying to do, because I cannot imagine why that would be desirable.

**Reply by rowlesmr** [reply]
I'm modelling xray scattering factors as sums of gaussians.

Y(s) = c + sum(a_i * exp(-b_i * s **2, i=1..5)

The order of the values of a and b don't matter (as long as the correct a goes with the correct b), and you gain additional insight if the parameters are listed in increasing b value.

**Comment by newville** [comment, marked as answer]
@rowlesmr I still think the answer is No. From a technical perspective, you certainly could not change the order of the Parameters and expect anything to work afterward.

**Comment by rowlesmr** [comment]
No worries. I'll work on making more robust post processing on my end.

Thanks
